File: src/net/fred/feedex/utils/PrefUtils.java
/**
 * FeedEx
 *
 * Copyright (c) 2012-2013 Frederic Julian
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package net.fred.feedex.utils;

import android.content.SharedPreferences;
import android.content.SharedPreferences.OnSharedPreferenceChangeListener;
import android.preference.PreferenceManager;

import net.fred.feedex.MainApplication;

public class PrefUtils {

    public static final String FIRST_OPEN = "FIRST_OPEN";

    public static final String IS_REFRESHING = "IS_REFRESHING";

    public static final String REFRESH_INTERVAL = "refresh.interval";
    public static final String REFRESH_ENABLED = "refresh.enabled";
    public static final String REFRESH_ON_OPEN_ENABLED = "refreshonopen.enabled";
    public static final String REFRESH_WIFI_ONLY = "refreshwifionly.enabled";

    public static final String NOTIFICATIONS_ENABLED = "notifications.enabled";
    public static final String NOTIFICATIONS_RINGTONE = "notifications.ringtone";
    public static final String NOTIFICATIONS_VIBRATE = "notifications.vibrate";
    public static final String NOTIFICATIONS_LIGHT = "notifications.light";

    public static final String LIGHT_THEME = "lighttheme";
    public static final String DISPLAY_IMAGES = "display_images";
    public static final String PRELOAD_IMAGE_MODE = "preload_image_mode";
    public static final String DISPLAY_OLDEST_FIRST = "display_oldest_first";
    public static final String DISPLAY_ENTRIES_FULLSCREEN = "display_entries_fullscreen";

    public static final String PROXY_ENABLED = "proxy.enabled";
    public static final String PROXY_PORT = "proxy.port";
    public static final String PROXY_HOST = "proxy.host";
    public static final String PROXY_WIFI_ONLY = "proxy.wifionly";
    public static final String PROXY_TYPE = "proxy.type";

    public static final String KEEP_TIME = "keeptime";

    public static final String FONT_SIZE = "fontsize";

    public static final String LAST_SCHEDULED_REFRESH = "lastscheduledrefresh";

    public static final String SHOW_READ = "show_read";

    public static boolean getBoolean(String key, boolean defValue) {
        SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(MainApplication.getContext());
        return settings.getBoolean(key, defValue);
    }

    public static void putBoolean(String key, boolean value) {
        SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(MainApplication.getContext()).edit();
        editor.putBoolean(key, value);
        editor.commit();
    }

    public static int getInt(String key, int defValue) {
        SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(MainApplication.getContext());
        return settings.getInt(key, defValue);
    }

    public static void putInt(String key, int value) {
        SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(MainApplication.getContext()).edit();
        editor.putInt(key, value);
        editor.commit();
    }

    public static long getLong(String key, long defValue) {
        SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(MainApplication.getContext());
        return settings.getLong(key, defValue);
    }

    public static void putLong(String key, long value) {
        SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(MainApplication.getContext()).edit();
        editor.putLong(key, value);
        editor.commit();
    }

    public static String getString(String key, String defValue) {
        SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(MainApplication.getContext());
        return settings.getString(key, defValue);
    }

    public static void putString(String key, String value) {
        SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(MainApplication.getContext()).edit();
        editor.putString(key, value);
        editor.commit();
    }

    public static void remove(String key) {
        SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(MainApplication.getContext()).edit();
        editor.remove(key);
        editor.commit();
    }

    public static void registerOnPrefChangeListener(OnSharedPreferenceChangeListener listener) {
        try {
            PreferenceManager.getDefaultSharedPreferences(MainApplication.getContext()).registerOnSharedPreferenceChangeListener(listener);
        } catch (Exception ignored) { // Seems to be possible to have a NPE here... Why??
        }
    }

    public static void unregisterOnPrefChangeListener(OnSharedPreferenceChangeListener listener) {
        try {
            PreferenceManager.getDefaultSharedPreferences(MainApplication.getContext()).unregisterOnSharedPreferenceChangeListener(listener);
        } catch (Exception ignored) { // Seems to be possible to have a NPE here... Why??
        }
    }
}

File: app/src/main/java/net/frju/flym/data/utils/PrefUtils.kt
/*
 * Copyright (c) 2012-2018 Frederic Julian
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http:></http:>//www.gnu.org/licenses/>.
 */

package net.frju.flym.data.utils

import android.content.SharedPreferences.OnSharedPreferenceChangeListener
import androidx.core.content.edit
import net.frju.flym.App
import org.jetbrains.anko.defaultSharedPreferences


object PrefUtils {

	const val FIRST_OPEN = "first_open"

	const val IS_REFRESHING = "is_refreshing"

	const val REFRESH_ENABLED = "refresh_enabled"
	const val REFRESH_INTERVAL = "refresh_interval"
	const val REFRESH_WIFI_ONLY = "refresh_wifi_only"

	const val DISPLAY_IMAGES = "display_images"

	const val PRELOAD_IMAGE_MODE = "preload_image_mode"
	const val PRELOAD_IMAGE_MODE__WIFI_ONLY = "WIFI_ONLY_PRELOAD"
	const val PRELOAD_IMAGE_MODE__ALWAYS = "ALWAYS_PRELOAD"

	const val KEEP_TIME = "keep_time"

	const val FONT_SIZE = "font_size"

	fun getBoolean(key: String, defValue: Boolean) =
			App.context.defaultSharedPreferences.getBoolean(key, defValue)

	fun putBoolean(key: String, value: Boolean) {
		App.context.defaultSharedPreferences.edit { putBoolean(key, value) }
	}

	fun getInt(key: String, defValue: Int) =
			App.context.defaultSharedPreferences.getInt(key, defValue)

	fun putInt(key: String, value: Int) {
		App.context.defaultSharedPreferences.edit { putInt(key, value) }
	}

	fun getLong(key: String, defValue: Long) =
			App.context.defaultSharedPreferences.getLong(key, defValue)

	fun putLong(key: String, value: Long) {
		App.context.defaultSharedPreferences.edit { putLong(key, value) }
	}

	fun getString(key: String, defValue: String) =
			App.context.defaultSharedPreferences.getString(key, defValue)

	fun putString(key: String, value: String) {
		App.context.defaultSharedPreferences.edit { putString(key, value) }
	}

	fun getStringSet(key: String, defValue: MutableSet<String>) =
			App.context.defaultSharedPreferences.getStringSet(key, defValue)

	fun putStringSet(key: String, value: MutableSet<String>) {
		App.context.defaultSharedPreferences.edit { putStringSet(key, value) }
	}

	fun remove(key: String) {
		App.context.defaultSharedPreferences.edit { PrefUtils.remove(key) }
	}

	fun registerOnPrefChangeListener(listener: OnSharedPreferenceChangeListener) {
		try {
			App.context.defaultSharedPreferences.registerOnSharedPreferenceChangeListener(listener)
		} catch (ignored: Exception) { // Seems to be possible to have a NPE here... Why??
		}

	}

	fun unregisterOnPrefChangeListener(listener: OnSharedPreferenceChangeListener) {
		try {
			App.context.defaultSharedPreferences.unregisterOnSharedPreferenceChangeListener(listener)
		} catch (ignored: Exception) { // Seems to be possible to have a NPE here... Why??
		}

	}
}
